src/lib/utilities/decode-payload.ts

```diff
@@ -1,31 +1,57 @@
-import type {
-  WorkflowExecutionStartedEventAttributes,
-  WorkflowExecutionCompletedEventAttributes,
-} from '$types';
-
-type DecodeEvent = BaseEvent & {
-  workflowExecutionStartedEventAttributes?: WorkflowExecutionStartedEventAttributes;
-  workflowExecutionCompletedEventAttributes?: WorkflowExecutionCompletedEventAttributes;
-};
+import type { Payload } from '$types';
+import { dataConverterPort } from '$lib/stores/data-converter-config';
+import { dataConverterWebsocket } from '$lib/stores/data-converter-websocket';
+import { get } from 'svelte/store';
+import { convertSinglePayloadDataConverter } from '$lib/services/data-converter';
+
+export const convertPayloadToJson = async (
+  eventAttribute: EventAttribute,
+): Promise<EventAttribute> => {
+  if (!eventAttribute) return eventAttribute;
+
+  // This anyAttribues allows us to use ?. notation to introspect the object which is a safe access pattern.
+  // Because of the way we wrote our discrimited union we have to use this any. If we have two objects that
+  // don't have the same property TS won't let us access that object without verifying the type string like
+  // attributes.type === "ATypeWithInput/Result"
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const anyAttributes = eventAttribute as any;
 
-export const convertPayloadToJson = (event: DecodeEvent): string => {
-  if (!event) return;
+  const port = get(dataConverterPort);
 
-  if (event.workflowExecutionStartedEventAttributes) {
-    const input = event.workflowExecutionStartedEventAttributes.input;
-    if (input === null) return 'null';
-    if (input) {
-      const [data] = input.payloads.map((payload) => payload.data);
-      return window.atob(String(data));
-    }
+  const potentialPayload = (anyAttributes?.input?.payloads ??
```

Comment:
This feels like it needs to be broken out into a utility function with a test.

Reply:
Yeah this entire function needs to become a well tested function if it's working to satisfaction I'll focus on that tomorrow